disable.ps1

```diff
@@ -148,150 +69,388 @@ function Resolve-ZenyaErrorMessage {
+                foreach ($newProperty in $newProperties) {
+                    # Add department as extension property to account body
+                    if ($newProperty.Name -eq "Department") {
+                        $accountBody.operations += @{ 
+                            op    = "replace"
+                            path  = "urn:ietf:params:scim:schemas:extension:enterprise:2.0:User:Department"
+                            value = $newProperty.Value
+                        }
+                    }
+                    # Add manager as extension property to account body
+                    elseif ($newProperty.Name -eq "Manager") {
+                        $accountBody.operations += @{ 
+                            op    = "replace"
+                            path  = "urn:ietf:params:scim:schemas:extension:enterprise:2.0:User:Manager"
+                            value = $newProperty.Value
+                        }
+                    }
+                    # Transform and add emails to account body
+                    elseif ($newProperty.Name -eq "Emails") {
+                        foreach ($email in $newProperty.Value) {
+                            if ($email.StartsWith("work:")) {
+                                $accountBody.operations += @{      
+                                    op    = "replace"
+                                    path  = "emails[type eq `"work`"].value"
+                                    value = $email -replace "work:", ""
+                                }
+                            }
+                        }
+                    }
+                    else {
+                        $accountBody.operations += @{
+                            op    = "replace"
+                            path  = $newProperty.name
+                            value = $newProperty.value
+                        }
+                    }
+                }
 
-    # Optionally return data for use in other systems
-    ExportData = [PSCustomObject]@{
-        id       = $aRef.id
-        userName = $aRef.userName
+                $body = ($accountBody | ConvertTo-Json -Depth 10)
+                $splatWebRequest = @{
+                    Uri             = "$baseUrl/scim/users/$($currentAccount.Id)"
+                    Headers         = $headers
+                    Method          = 'PATCH'
+                    Body            = ([System.Text.Encoding]::UTF8.GetBytes($body))
+                    ContentType     = "application/json;charset=utf-8"
+                    UseBasicParsing = $true
+                }
+
+                if (-Not($actionContext.DryRun -eq $true)) {
+                    Write-Verbose "Updating account [$($currentAccount.Username)]. AccountReference: $($actionContext.References.Account | ConvertTo-Json -Depth 10). Body: $($updateAccount | ConvertTo-Json -Depth 10)"
+        
+                    $updatedAccount = Invoke-RestMethod @splatWebRequest -Verbose:$false
```

Comment:
I don't see where the account is disabled. Is this done in the UI mapping? If so, then we need to know how it should be mapped. 
Also I expect in the disable event to have an auditlog about disabling the user account.

Reply:
This indeed is in the UI mapping, for the field "Active". The mapping examples will be added to the readme.
The logging has been changed from update to disable messages.